### PR TITLE
bgp: extract exportPodCIDRReconciler logic into a generic function

### PR DESCRIPTION
**Pull request by jibi**
This commit extracts most of the exportPodCIDRReconciler logic into a
more generic function, exportAdvertisementsReconciler, so that it can be
reused by other subsystems which only need to advertise a set of IPNets.

### CODE REVIEW

File: pkg/bgpv1/gobgp/advertisements_reconciler.go [hunk 1]
```diff
@@ -112,48 +110,47 @@ func exportPodCIDRReconciler(ctx context.Context, _ *BGPRouterManager, sc *Serve
...
+		// cidrs to leave advertised.
 		if m.b && m.a {
 			toKeep = append(toKeep, *m.advrt)
 		}
 	}
 
 	if len(toAdvertise) == 0 && len(toWithdraw) == 0 {
 		l.Debugf("No reconciliation necessary")
-		return nil
+		return params.currentAdvertisements, nil
```
Comment (lmb) [inline pkg/bgpv1/gobgp/advertisements_reconciler.go]: Should this return a copy of the slice?

**Review by lmb — APPROVED** [review]

**Comment by jibi** [comment]
/test-1.16-4.19

**Review by dylandreimerink — APPROVED** [review]

**Pull request merged**